snap/hooks/configure

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env python3
+
+import sys
+
+from charmcraft import snap
```

Comment:
I would personally move all the logic in charmcraft.snap here. You can import snap.hooks.configure if you wanted from pytest (it may be inconvenient if the root is moved though). In any case, I would rename snap to snap_hooks or something similar.

Reply:
The logic in charmcraft.snap is meant to be used by the charmcraft application at runtime to get/validate the snap configuration (to actually use it).

Reply:
@cjp256 good point!